include/nbl/core/IReferenceCounted.h

```diff
@@ -62,7 +62,7 @@ class NBL_FORCE_EBO IReferenceCounted : public Interface, public AllocationOverr
 		You will not have to drop the pointer to the loaded texture,
 		because the name of the method does not start with 'create'.
 		The texture is stored somewhere by the driver. */
-		inline void grab() const { ReferenceCounter++; }
+		inline std::atomic_uint grab() const { return ReferenceCounter++; }

```

Comment:
you're not returning an atomic, just a uint32_t, because there's no atomicity after `++` returns (the value is immediately out of date)